poetry/masonry/publishing/uploader.py

```diff
@@ -228,6 +229,11 @@ def _upload_file(self, session, url, file):
                 encoder, lambda monitor: bar.set_progress(monitor.bytes_read)
             )
 
+            # Check for a redirect first
+            resp = session.head(url)
+            if resp.status_code == status_codes.codes.moved:
```

Comment:
This approach means we'd have to make two requests every time, even when there is no redirect. Wouldn't we be able to just try the endpoint and then follow the redirect if necessary?

Reply:
That's what I did originally, but that means POSTing the package data twice, so anyone who configured the wrong URL for their repo will have a big perf hit. On the other hand, a HEAD request is very cheap and isn't noticeable. Happy to do some benchmarks if you are not convinced.

Reply:
That's a good point that it would be expensive to POST twice.